hello-robot/stretch_ros2
# Update demos to work with HelloNode

This PR makes several changes to the stretch_demos package:
1. Updates HelloNode class by creating node through the Node class - previously done with rclpy.create_node method
2. Updates all demos to subclass from the HelloNode class to remove repetitive code
3. Updates the grasp object demo to work with a DexWrist
4. Adds instructions to run keyboard teleop in a separate terminal for each demo

To test, follow instructions in the README

Bugs that require further investigation:
1. Plane detection in the clean surface demo is poor because of poor fill rate from RealSense. The pointcloud feed in the realsense-viewer has a much better fill rate than one generated by the ROS drivers.
2. The open drawer demo applies too much effort when descending the hook to grab the drawer handle. This is because of the contact threshold bug in the lift. See https://github.com/hello-robot/stretch_firmware/pull/22
